tests/rustdoc-json/trait_alias.rs

```diff
@@ -0,0 +1,18 @@
+// ignore-tidy-linelength
+#![feature(trait_alias)]
+
+// @set StrLike = "$.index[*][?(@.name=='StrLike')].id"
+// @is "$.index[*][?(@.name=='StrLike')].visibility" \"public\"
+// @has "$.index[*][?(@.name=='StrLike')].inner.trait_alias"
+// @is "$.index[*][?(@.name=='StrLike')].span.filename" $FILE
+pub trait StrLike = AsRef<str>;
+
+// @is "$.index[*][?(@.name=='f')].inner.function.decl.output.impl_trait[0].trait_bound.trait.id" $StrLike
+pub fn f() -> impl StrLike {
+    "heya"
+}
+
+// @!is "$.index[*][?(@.name=='g')].inner.function.decl.output.impl_trait[0].trait_bound.trait.id" $StrLike
```

Comment:
Why does this test pass?? the user didn't write `StrLike` here?

Reply:
Yes, I don't know if it even makes sense to test this from the perspective of the implementation of rustc/rustdoc, but who knows, maybe someday it'll accidentally start conflating trait aliases and their equivalent trait, then this test case will heroically prevent a disaster

Reply:
Oh - I am confused. I missed the `!`.

Reply:
That is to say, I thought this was testing *for* the trait alias, not testing for the *lack* of it.